s3_parq/s3_parq.py

```diff
@@ -9,23 +10,37 @@ class S3Parq:
     def __init__(self, **kwargs):
         self._set_kwargs_as_attrs(**kwargs)
 
-    def publish(self, partitions:iter=set())->None:
+    def publish(self, dataset:str=None, bucket:str=None, dataframe:pd.DataFrame=None, key_prefix:str=None, partitions:iter=None)->None:
+        ## this round-about setting of args gives IDEs back hinting and adds mechanical type checking. 
+        for attr, val in {"dataset":dataset, "bucket":bucket, "dataframe":dataframe, "partitions":partitions, "key_prefix":key_prefix}.items():
+            if val is not None:
+                self._set_kwargs_as_attrs({attr:val})
+
         required_attributes = ('dataset', 'bucket', 'dataframe',)
         self._check_required_attr(required_attributes)
 
-        publish = S3PublishParq(dataset=self.dataset,
-                                bucket=self.bucket,
-                                dataframe=self.dataframe,
-                                key_prefix=getattr(self,"key_prefix",''),
-                                partitions=partitions
-                                )
-        return publish.do_publish()
-
-    def fetch(self)->None:
+        S3PublishParq(  dataset=self.dataset,
+                        bucket=self.bucket,
+                        dataframe=self.dataframe,
+                        key_prefix=getattr(self,"key_prefix",''),
+                        partitions=getattr(self,"partitions",[])
+                     )
+
+    def fetch(self, dataset:str=None, bucket:str=None, filters:dict=None, key_prefix:str=None)->None:
```

Comment:
@schafrn how does this feel, better? or still too much magic? 
lmk and I'll take another pass if needed.

Reply:
Better, thank you!